commandapi-core/src/main/java/dev/jorel/commandapi/executors/CommandArguments.java

```diff
@@ -10,49 +10,18 @@
 
 /**
  * This class stores the arguments for this command
+ *
+ * @param args      The arguments for this command
+ * @param argsMap   The arguments for this command mapped to their node names. This is an ordered map
+ * @param rawArgs   The raw arguments for this command.
+ * @param rawArgsMap   The raw arguments for this command mapped to their node names. This is an ordered map
+ * @param fullInput The command string a player has entered (including the /)
  */
 @SuppressWarnings("unchecked")
-public class CommandArguments {
-
-	private final Object[] args;
-	private final String[] rawArgs;
-	private final Map<String, Object> argsMap;
-	private final Map<String, String> rawArgsMap;
-	private final String fullInput;
-
-	/**
-	 * Constructs a new CommandArguments instance
-	 *
-	 * @param args      The arguments for this command
-	 * @param argsMap   The arguments for this command mapped to their node names. This is an ordered map
-	 * @param rawArgs   The raw arguments for this command.
-	 * @param rawArgsMap   The raw arguments for this command mapped to their node names. This is an ordered map
-	 * @param fullInput The raw command a player has entered
-	 */
-	public CommandArguments(Object[] args, Map<String, Object> argsMap, String[] rawArgs, Map<String, String> rawArgsMap, String fullInput) {
-		this.args = args;
-		this.rawArgs = rawArgs;
-		this.argsMap = argsMap;
-		this.rawArgsMap = rawArgsMap;
-		this.fullInput = fullInput;
-	}
+public record CommandArguments(Object[] args, Map<String, Object> argsMap, String[] rawArgs, Map<String, String> rawArgsMap, String fullInput) {
```

Comment:
JavaDocs for the parameters should be added to the individual arguments. See [SuggestionInfo](https://github.com/JorelAli/CommandAPI/blob/55fd6ffa45ca6f692a4237cbfada0247fc8a82c2/commandapi-core/src/main/java/dev/jorel/commandapi/SuggestionInfo.java#L37) as an example, where each parameter has JavaDocs attached to them.

I've discussed this in the past, but certain IDEs (Eclipse) doesn't generate JavaDocs for the function implementations of record fields (e.g. `args()` or `argsMap()`) unless JavaDocs are added to the given parameters.